stix_shifter/utils/error_response.py

```diff
@@ -0,0 +1,124 @@
+from requests.exceptions import SSLError, ConnectionError
+from enum import Enum
+import importlib
+import traceback
+from ..utils.error_mapper_base import ErrorMapperBase
+import collections
+import json
+
+class ErrorCode(Enum):
+    TRANSMISSION_UNKNOWN = 500
+    TRANSMISSION_CONNECT = 600
+    TRANSMISSION_AUTH_SSL = 610
+    TRANSMISSION_AUTH_CREDENTIALS = 620
+    TRANSMISSION_MODULE_DEFAULT_ERROR = 700
+    TRANSMISSION_QUERY_PARSING_ERROR = 802
+    TRANSMISSION_QUERY_LOGICAL_ERROR = 803
+    TRANSMISSION_RESPONSE_EMPTY_RESULT = 903
```

Comment:
How did you decide on error codes? i.e. starting at 802, 903, etc.

Reply:
5xx - transmission unexpected error
6xx - connectivity errors
7xx - submodule unhandled errors
8xx - "error in query" error
9xx - special case for remote system errors
10xx - non fatal-, retry-later-  errors

if you do not like them sometimes starting from xx[^0] - just let me know.

Reply:
I'm still confused with error codes? Why don't you..

```
    TRANSMISSION_UNKNOWN = 500
    TRANSMISSION_CONNECT = 600
    TRANSMISSION_AUTH_SSL = 601
    TRANSMISSION_AUTH_CREDENTIALS = 602
    TRANSMISSION_MODULE_DEFAULT_ERROR = 700
    TRANSMISSION_QUERY_PARSING_ERROR = 800
    TRANSMISSION_QUERY_LOGICAL_ERROR = 801
    TRANSMISSION_RESPONSE_EMPTY_RESULT = 900
    TRANSMISSION_SEARCH_DOES_NOT_EXISTS = 901
    TRANSMISSION_INVALID_PARAMETER = 902
    TRANSMISSION_REMOTE_SYSTEM_IS_UNAVAILABLE = 1000
```